### ISSUE TITLE
Want to be able to have deferred destroy of tasks and aios

### ISSUE THREAD

**Issue by gdamore** [nanomsg/nng#511]
There are instances where we want to deallocate stuff safely, but we can't "wait" for the operation to complete because we would be waiting on ourselves.

What would be nice would be to have a new API for "deferred" destroy, that didn't run synchronously, but scheduled a later reaping of the resource.  This could be called from within a callback on an aio to clean up the aio itself.

Note that there are many paradigms where this structure is a sign of poor coding/architecture, but there are situations where this would save effort.

An example of this sort of problem is discussed in #508 -- our own websocket demo suffered from this.
